grpc-spring-boot-starter/src/main/java/org/lognet/springboot/grpc/GRpcServerRunner.java

```diff
@@ -158,8 +158,8 @@ public void destroy() throws Exception {
 
                     if (!beansWithAnnotation.isEmpty()) {
                         return beansWithAnnotation.containsKey(name);
-                    } else if (beanDefinition.getSource() instanceof AnnotatedTypeMetadata) {
```

Comment:
May I ask you why you switched form `AnnotatedTypeMetadata` to `MethodMetadata` ?

Reply:
That is a good question. I think I mixed up the hierarchies ... Using `AnnotatedTypeMetadata` would make way more sense, but maybe there was an error with that. I will try it again tomorrow.